### ISSUE TITLE
Routes manifest only returns one route when 2+ routes have same component name

### ISSUE THREAD

**Issue by remjx** [blitz-js/blitz#2558]
### What is the problem?

When two pages exist that export `BlitzPage`s with the same name, e.g. `Page` in this example:

```
// app/pages/page1.tsx and app/pages/page2.tsx
const Page: BlitzPage = () => <Component />
export default Page
```

the route manifest only generates one entry because there is a naming conflict, causing one of the routes to be missing/overwritten in the manifest:

```
exports.Routes = {
  Page: (query) => ({ pathname: "/page2", query }),
}
```

This should cause an error during the build/dev process.

### Paste all your error logs here:

N/A

### Paste all relevant code snippets here:

```
PASTE_HERE (leave the ``` marks)
```

### What are detailed steps to reproduce this?

1.

### Run `blitz -v` and paste the output here:

```
Windows 10 | win32-x64 | Node: v14.17.1

blitz: 0.33.0 (global)
blitz: 0.38.2 (local)

  Package manager: npm
  System:
    OS: Windows 10 10.0.19042
    CPU: (6) x64 Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz
    Memory: 4.75 GB / 15.94 GB
  Binaries:
    Node: 14.17.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  npmPackages:
    @prisma/client: 2.24.1 => 2.24.1
    blitz: 0.38.2 => 0.38.2
    prisma: 2.24.1 => 2.24.1
    react: ^18.0.0-alpha-c96b78e0e => 18.0.0-alpha-ed6c091fe-20210701
    react-dom: ^18.0.0-alpha-c96b78e0e => 18.0.0-alpha-ed6c091fe-20210701
    typescript: ^4.3.2 => 4.3.5
```

**Comment by flybayer** [comment]
Good catch, thanks!

I believe we just need to add a check right before this code: https://github.com/blitz-js/blitz/blob/canary/packages/server/src/stages/route-import-manifest/route-import-manifest.ts#L187-L190

**Comment by ravindranrahul** [comment]
Hi can i take this up?

**Comment by abuuzayr** [comment]
absolutely, @ravindranrahul ! i'll assign it to you, do let us know if you need any help.

**Comment by ravindranrahul** [comment]
Hey @abuuzayr, Im pretty new to this. So far:

- I have setup my local blitz
- After yarn dev on root folder & blitz dev inside `examples/auth`, Im able to get the example app working.

However,  the following console log  in `packages/server/src/stages/route-import-manifest/route-import-manifest.ts` doesn't seem to work. 

![Screenshot from 2021-07-19 13-10-42](https://user-images.githubusercontent.com/10168946/126121676-104c9e48-5140-4f26-b6fd-578b8626ba2e.png)

I have tried restarting both yarn dev and blitz dev. No luck. Am I missing something?

**Comment by flybayer** [comment]
@ravindranrahul to run that code, run `blitz routes` :)

**Comment by flybayer** [comment]
sorry for the delayed response! You can always ask in the contributor section in discord too

**Comment by ravindranrahul** [comment]
@flybayer Thanks for getting back to me! Let me try that. I'll direct additional questions to discord group.

**Comment by ravindranrahul** [comment]
I have reused the `dedupeBy` method for highlighting duplicate paths. FYI -  While testing it shows `Test` is also a duplicate entry in `examples/auth`. Potentially because of conflict between `pages/ssr.tsx` & `pages/test.tsx`.

![Screenshot from 2021-07-28 20-18-30](https://user-images.githubusercontent.com/10168946/127345383-44bdbb01-309d-47b4-a0c1-929f3e9a0e64.png)

@flybayer  let me know whats the best way to throw a build error.
